_api/hasura/cron/giveOnchainSyncer.ts

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable no-console */
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { attestGiveOnchain } from '../../../api-lib/eas';
+import { order_by } from '../../../api-lib/gql/__generated__/zeus';
+import { adminClient } from '../../../api-lib/gql/adminClient';
+import { verifyHasuraRequestMiddleware } from '../../../api-lib/validate';
+
+const LIMIT = 10;
+
+async function handler(_req: VercelRequest, res: VercelResponse) {
+  const gives = await giveToSync();
+
+  console.log('Gives to write onchain: ', gives.length);
+  console.log({ gives });
+
+  try {
+    let errors = 0;
+    let success = 0;
+    for (const give of gives) {
+      try {
+        console.log({
+          giver: give.giver_profile_public?.address,
+          receiver: give.target_profile_public?.address,
+          skill: give.skill,
+        });
+        console.log('Writing give onchain for give id: ', give.id);
+        const { attestUid, txHash } = await attestGiveOnchain(give);
+
+        console.log(
+          'Success: attested give for give.id : ',
+          give.id,
+          ' receiver address: ',
+          give.target_profile_public?.address,
+          'attestUid: ',
+          attestUid
+        );
+
+        await adminClient.mutate(
+          {
+            update_colinks_gives_by_pk: [
+              {
+                _set: {
+                  attestation_uid: attestUid,
+                  tx_hash: txHash,
+                },
+                pk_columns: {
+                  id: give.id,
+                },
+              },
+              {
+                __typename: true,
+              },
+            ],
+          },
+          {
+            operationName: 'cron_giveOnchainSyncer__attestGive',
+          }
+        );
+
+        success++;
+      } catch (e: any) {
+        errors++;
+
+        await adminClient.mutate(
```

Comment:
should this just happen in the attest function so the updates are all in one place